app/controllers/posts_controller.rb

```diff
@@ -210,6 +210,13 @@ def update
       edit_reason: params[:post][:edit_reason]
     }
 
+    # Check if there is no edit conflict caused by two users saving different
+    # versions of the post.
+    raw_old = params[:post][:raw_old]
+    if !raw_old.empty? && raw_old != post.raw
```

Comment:
The "double negation" feels weird.

```suggestion
    if raw_old.present? && raw_old != post.raw
```